src/controller/LoginController.php

```diff
@@ -39,10 +41,19 @@ public function logoutAction() {
         }
     }
 
-    private function userExist($userName, $password) {
-        $sqlQuery = "SELECT * FROM users WHERE userName='$userName' AND password='$password'";
+    private function userExist($userName, $hashedAndSaltedPassword) {
+        $sqlQuery = "SELECT * FROM users WHERE userName='$userName' AND password='$hashedAndSaltedPassword'";
         $result = $this->connection->query($sqlQuery);
         return $result->num_rows > 0;
     }
 
+    private function hashAndSaltPassword($userName, $password) {
```

Comment:
Kolkas ignoruojam kodo logiką/saugumą ir pakalbam tiesiog apie struktūrą. `hashAndSaltPassword` kartojasi 2 kodo vietose, o kaip turbūt jau pats įtarei - to nenorime. `src` aplanke sukuriame naują aplanką `Util` (+ pervadiname `controller` į `Controller`), Jame sukuriame failą/klasę pvz `PasswordHasher` su vienu statiniu metodu `getPasswordHash` kurį ir panaudojame 2 skirtingose vietose. `dhfjdhfjhdsjfh` ir panašius darinius išskiriame kaip klasės konstantas